private/tools/java/rules/jvm/external/jar/ListPackages.java

```diff
@@ -0,0 +1,84 @@
+package rules.jvm.external.jar;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.SortedMap;
+import java.util.SortedSet;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class ListPackages {
+
+  private static final Predicate<String> IS_NUMERIC_VERSION = Pattern.compile("[1-9][0-9]*").asPredicate();
+
+  public static void main(String[] args) throws IOException {
+    if (args.length != 2 || !"--argsfile".equals(args[0])) {
+      System.err.printf("Required args: --argsfile /path/to/argsfile%n");
+      System.exit(1);
+    }
+
+    SortedMap<String, SortedSet<String>> index = new TreeMap<>();
+
+    Files.lines(Paths.get(args[1]))
+        .parallel()
```

Comment:
Since this is doing disk IO, does being in parallel make much difference?

Reply:
It does! After adding the `BufferedInputStream`, this provided an improvement for a large sample repo I have from in the 5-6 second range down to in the 1-2 second range. (Before adding the `BufferedInputStream` it was even bigger; 12-13 seconds -> 1-3 seconds)

That said, I've rewritten how the parallel collection happens, because parallel writes to a non-concurrent-safe collection were causing the CI failures - this now does a `collect` rather than writes from threads.